src/lib/containers/drug_upload_tool/UserFileGrid.tsx

```diff
@@ -0,0 +1,267 @@
+// Alina TODO:
+//  - pagination and scroll
+// - submission
+
+import './DrugUploadTool.scss'
+
+import * as React from 'react'
+import { Link } from 'react-router-dom'
+
+import { Entity, } from '../../utils/jsonResponses/Entity'
+import { EntityLookupRequest } from '../../utils/jsonResponses/EntityLookupRequest'
+import { UserProfile } from '../../utils/jsonResponses/UserProfile'
+import { SynapseClient } from '../../utils'
+import { faTrash } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import WarningModal from './WarningModal'
+
+import moment from 'moment';
+
+type UserData = {
+  fileList: Entity[];
+  existingData: any;
+  dataFolderId: string;
+}
+
+export type UserFileGridProps = {
+  token?: string;
+  parentContainerId: string; //projectId
+  pathpart: string;
+}
+
+type UserFileGridState = {
+  token?: string;
+  userProfile?: any;
+  dataFolderId?: string,
+  fileList: Entity[]
+  isLoading: boolean;
+  modalContext?: { 'action': Function, 'arguments': any[] }
+}
+
+
+export default class UserFileGrid extends React.Component<UserFileGridProps, UserFileGridState> {
+  modalTitle = 'Trash Submission?'
+  modalCopy = <><p>This submission is currently incomplete and has not been submitted. If you trash this submission, you won't be able to recover the data.</p>
+    <p>Are you sure you want to trash this submission?' </p></>
+
+  constructor(props: UserFileGridProps) {
+    super(props)
+    this.state = {
+      isLoading: true,
+      token: this.props.token,
+      fileList: []
+    }
+  }
+
+  async componentDidMount(){
+    await this.refresh(this.state.token)
+  }
+
+  async componentDidUpdate(prevProps: any) {
+    const shouldUpdate = this.props.token !== prevProps.token
+    if (shouldUpdate) {
+      await this.refresh(this.props.token)
+    }
+  }
+
+  async refresh (token?: string) {
+    this.setState({ token })
```

Comment:
I would remove token from state and only use `this.props.token`, copying props to state is an anti-pattern, (with a few exceptions)